src/main/scala/com/raphtory/core/analysis/GraphLens.scala

```diff
@@ -48,29 +49,32 @@ final case class GraphLens(
     viewTimer.update(System.currentTimeMillis() - startTime)
     result
   }
-
+  def getVertices(): ParIterable[Vertex] = {
+    vertexCount.set(vertexMap.size)
+    vertexMap.map(x=>x._2)
+  }
   def getMessagedVertices(): ParIterable[Vertex] = {
     val startTime = System.currentTimeMillis()
-    val result    = getVertices.filter(_.hasMessage())
-    messageFilter = true
+    val result    = vertexMap.collect {
+      case (k, vertex) if vertex.hasMessage() => vertex
+    }
     viewTimer.update(System.currentTimeMillis() - startTime)
+    vertexCount.set(result.size)
     result
   }
 
-  def getVertices(): ParIterable[Vertex] = vertexMap.map(x=>x._2)
-
-  def checkVotes(): Boolean =
```

Comment:
This was a pretty poorly optimised function and also would have been returning the wrong results as after the vertex pulls its queue it would have been emptied.